File: prev_exams_and_ans/2016ccc/q5_tandem_bicycle/TandemBicycle.java
import java.io.IOException;
import java.util.Scanner;
import java.util.Arrays;

public class TandemBicycle {

	public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);
		int maxMinIndicator = sc.nextInt();
        int N = sc.nextInt();
        
        int dmo[] = new int[N];
        int peg[] = new int[N];

        for (int i = 0; i < N; i++)
            dmo[i] = sc.nextInt();

        for (int i = 0; i < N; i++)
            peg[i] = sc.nextInt();

        sc.close();

        Arrays.sort(dmo);
        Arrays.sort(peg);

		// if (maxMinIndicator == 2) {
		// 	System.out.println(maxSpd(dmo, peg));
		// } else if (maxMinIndicator == 1) {
		// 	System.out.println(minSpd(dmo, peg));	
        // }
        System.out.println(totalSpd(dmo, peg, maxMinIndicator));
    }
    
    public static int maxSpd(int[] arr1, int[] arr2) {
        int sum = 0;
        int arrLen = arr1.length;
        for (int i = 0; i < arrLen; i++) {
            sum += Math.max(arr1[i], arr2[arrLen-1-i]);
        }
        return sum;
    }

    public static int minSpd(int[] arr1, int[] arr2) {
        int sum = 0;
        int arrLen = arr1.length;
        for (int i = 0; i < arrLen; i++) {
            sum += Math.max(arr1[i], arr2[i]);
        }
        return sum;
    }

    public static int totalSpd(int[] arr1, int[] arr2, int indicator) {
        int sum = 0;
        int arrLen = arr1.length;
        for (int i = 0; i < arrLen; i++) {
            sum += Math.max(arr1[i], arr2[(indicator == 1) ? i : (arrLen - i - 1)]);
        }
        return sum;
    }
}

File: prev_exams_and_ans/2016ccc/q5_tandem_bicycle/tandem_bicycle.py
import sys
input = sys.stdin.readline
max_min_indicator = int(input())
N = int(input()) # useless

dmo = [int(x) for x in input().split()]
peg = [int(x) for x in input().split()]

dmo.sort()
peg.sort()

def max_spd(arr1, arr2):
    sum = 0
    arr_len = len(arr1)
    for i in range(0, arr_len):
        sum += max(arr1[i], arr2[arr_len-1-i])
    return sum

def min_spd(arr1, arr2):
    sum = 0
    arr_len = len(arr1)
    for i in range(0, arr_len):
        sum += max(arr1[i], arr2[i])
    return sum

print(max_spd(dmo, peg) if max_min_indicator == 2 else min_spd(dmo, peg))